tests/utils/queryApiGateway.test.tsx

```diff
@@ -195,13 +195,31 @@ describe('Building the API url', () => {
 // Test getUniqueNumber()
 describe('The unique number', () => {
   it('is returned when given the correct ID key', () => {
-    const idHeaderName = 'id'
-    const id = getUniqueNumber(goodRequest, idHeaderName)
+    // Mock process.env
+    const restore = mockEnv({
+      ID_HEADER_NAME: 'id',
+    })
+
+    const id = getUniqueNumber(goodRequest)
     expect(id).toStrictEqual(goodRequest.headers.id)
+
+    // Restore env vars
+    restore()
   })
 
-  it('is not case sensitive', () => {
-    const key = 'uniqueNumber'
+  // Each test case should be:
+  // [test description, mock env var
+  const caseSensitiveTestCases = [
+    ['lowercase', 'uniquenumber'],
+    ['camelCase', 'uniqueNumber'],
+    ['uppercase', 'UNIQUENUMBER'],
+  ]
+  it.each(caseSensitiveTestCases)('is not case sensitive: %s', (description: string, key: string) => {
+    // Mock process.env
+    const restore = mockEnv({
+      ID_HEADER_NAME: key,
+    })
+
     const request = {
       headers: {
         uniquenumber: 'lower',
```

Comment:
Should this be using `key` or `id`? I don't quite follow the setup steps of setting the env, then building the request, then running the test.

Reply:
This test builds a different `request` object, instead of using `goodRequest` like in the other tests in this file. This test ensures that the `ID_HEADER_NAME` environment variable is **not** case sensitive. The setup for the test sets up the `ID_HEADER_NAME` env var with three different **case-sensitive** versions of the string `uniquenumber` and ensures that they all have the same behavior.

Reply:
nice!